src/plugins/discover/public/context_awareness/profile_providers/observability/observability_root_profile/accessors/get_app_menu.ts

```diff
@@ -0,0 +1,161 @@
+        registerDatasetQualityLink(registry, services);
+        // Register alerts sub menu actions
+        registerCreateSLOAction(registry, services, params);
+        registerCustomThresholdRuleAction(registry, services, params);
+
+        return prevValue.appMenuRegistry(registry);
+      },
+    };
+  };
+
+const registerDatasetQualityLink = (
+  registry: AppMenuRegistry,
+  { share, timefilter }: ProfileProviderServices
+) => {
+  const dataQualityLocator =
+    share?.url.locators.get<DataQualityLocatorParams>(DATA_QUALITY_LOCATOR_ID);
+
+  if (dataQualityLocator) {
+    registry.registerCustomAction({
+      id: 'dataset-quality-link',
+      type: AppMenuActionType.custom,
+      controlProps: {
+        label: i18n.translate('discover.observabilitySolution.appMenu.datasets', {
+          defaultMessage: 'Data sets',
+        }),
+        testId: 'discoverAppMenuDatasetQualityLink',
+        onClick: ({ onFinishAction }) => {
+          const refresh = timefilter.getRefreshInterval();
+          const { from, to } = timefilter.getTime();
+
+          dataQualityLocator.navigate({
+            filters: {
+              timeRange: {
+                from: from ?? 'now-24h',
+                to: to ?? 'now',
+                refresh,
+              },
+            },
+          });
+
+          onFinishAction();
+        },
+      },
+    });
+  }
+};
+
+const registerCustomThresholdRuleAction = (
+  registry: AppMenuRegistry,
+  { data, triggersActionsUi }: ProfileProviderServices,
+  { dataView }: AppMenuExtensionParams
+) => {
+  registry.registerCustomActionUnderSubmenu(AppMenuActionId.alerts, {
+    id: AppMenuActionId.createRule,
+    type: AppMenuActionType.custom,
+    order: 101,
+    controlProps: {
+      label: i18n.translate('discover.observabilitySolution.appMenu.customThresholdRule', {
+        defaultMessage: 'Create custom threshold rule',
+      }),
+      iconType: 'visGauge',
+      testId: 'discoverAppMenuCustomThresholdRule',
+      onClick: ({ onFinishAction }) => {
+        const index = dataView?.toMinimalSpec();
+        const { filters, query } = data.query.getState();
+
+        return triggersActionsUi.getAddRuleFlyout({
+          consumer: 'logs',
+          ruleTypeId: OBSERVABILITY_THRESHOLD_RULE_TYPE_ID,
+          canChangeTrigger: false,
+          initialValues: {
+            params: {
+              searchConfiguration: {
+                index,
+                query,
+                filter: filters,
+              },
+            },
+          },
+          onClose: onFinishAction,
+        });
+      },
+    },
+  });
+};
+
+const registerCreateSLOAction = (
+  registry: AppMenuRegistry,
+  { data, discoverShared }: ProfileProviderServices,
+  { dataView, isEsqlMode }: AppMenuExtensionParams
+) => {
+  const sloFeature = discoverShared.features.registry.getById('observability-create-slo');
+
+  if (sloFeature) {
+    registry.registerCustomActionUnderSubmenu(AppMenuActionId.alerts, {
```

Comment:
SLOs also don't seem to support ES|QL, and I wonder if they should still be available in ES|QL mode?
![CleanShot 2024-11-08 at 00 09 41@2x](https://github.com/user-attachments/assets/4756de69-505d-4420-9cea-ddd29ff596f1)

Reply:
Same reason as for the threshold rule, I feel it's still beneficial to have a direct line to SLOs for the currently explored source